### PR TITLE
Migrate out_copy and out_roundrobin to v0.14 API

### LINKED ISSUE #1040: unexpected error error="undefined method `router=' for #<Fluent::Plugin::StdoutOutput:…>
Masa told me: 

> This is a bug of out_copy and non-emittable output combination, e.g.
> out_stdout. v0.14.1 will fix this problem.

As per his request I open an issue about it.

With 

    # gem list | grep fluent
    fluent-mixin-config-placeholders (0.4.0)
    fluent-plugin-elasticsearch (1.5.0)
    fluent-plugin-parser (0.6.1)
    fluent-plugin-secure-forward (0.4.1)
    fluent-plugin-snmptrap (0.0.1)
    fluentd (0.14.0, 0.12.24)
    fluentd-ui (0.4.3)

and

```
<match *.**>
  @type copy
  <store>
    @type elasticsearch
    host localhost
    port 9200
    include_tag_key true
    tag_key @log_name
    logstash_format true
    flush_interval 10s
  </store>
  <store>
    @type stdout
  </store>
</match>
```

I get

```
2016-06-07 12:52:40 +0200 [info]: reading config file path="/etc/fluent/fluent.conf"
2016-06-07 12:52:40 +0200 [info]: starting fluentd-0.14.0
2016-06-07 12:52:40 +0200 [info]: spawn command to main: /usr/bin/ruby2.1 -Eascii-8bit:ascii-8bit /usr/local/bin/fluentd -c /etc/fluent/fluent.conf --no-super
visor
2016-06-07 12:52:41 +0200 [info]: reading config file path="/etc/fluent/fluent.conf"
2016-06-07 12:52:41 +0200 [info]: starting fluentd-0.14.0 without supervision
2016-06-07 12:52:41 +0200 [info]: gem 'fluent-mixin-config-placeholders' version '0.4.0'
2016-06-07 12:52:41 +0200 [info]: gem 'fluent-plugin-elasticsearch' version '1.5.0'
2016-06-07 12:52:41 +0200 [info]: gem 'fluent-plugin-parser' version '0.6.1'
2016-06-07 12:52:41 +0200 [info]: gem 'fluent-plugin-secure-forward' version '0.4.1'
2016-06-07 12:52:41 +0200 [info]: gem 'fluent-plugin-snmptrap' version '0.0.1'
2016-06-07 12:52:41 +0200 [info]: gem 'fluentd' version '0.14.0'
2016-06-07 12:52:41 +0200 [info]: gem 'fluentd' version '0.12.24'
2016-06-07 12:52:41 +0200 [info]: adding match pattern="system.syslog.mail.info" type="parser"
2016-06-07 12:52:41 +0200 [info]: adding match pattern="debug.**" type="stdout"
2016-06-07 12:52:41 +0200 [info]: adding match pattern="*.**" type="copy"
2016-06-07 12:52:41 +0200 [error]: unexpected error error="undefined method `router=' for #<Fluent::Plugin::StdoutOutput:007ff1bfc1ba48>"
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/plugin/out_copy.rb:47:in `block in configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/plugin/out_copy.rb:39:in `each'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/plugin/out_copy.rb:39:in `configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/agent.rb:135:in `add_match'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/agent.rb:69:in `block in configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/agent.rb:63:in `each'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/agent.rb:63:in `configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/root_agent.rb:86:in `configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/engine.rb:119:in `configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/engine.rb:93:in `run_configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/supervisor.rb:576:in `run_configure'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/supervisor.rb:381:in `block in run_worker'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/supervisor.rb:509:in `call'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/supervisor.rb:509:in `main_process'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/supervisor.rb:378:in `run_worker'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/lib/fluent/command/fluentd.rb:266:in `<top (required)>'
  2016-06-07 12:52:41 +0200 [error]: /usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:73:in `require'
  2016-06-07 12:52:41 +0200 [error]: /usr/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:73:in `require'
  2016-06-07 12:52:41 +0200 [error]: /var/lib/gems/2.1.0/gems/fluentd-0.14.0/bin/fluentd:5:in `<top (required)>'
  2016-06-07 12:52:41 +0200 [error]: /usr/local/bin/fluentd:23:in `load'
  2016-06-07 12:52:41 +0200 [error]: /usr/local/bin/fluentd:23:in `<main>'
2016-06-07 12:52:42 +0200 [info]: Worker 0 finished unexpectedly with status 1
```

### PR DESCRIPTION
**Pull request by tagomoris**
This change fixes #1040 too.

### CODE REVIEW

**Comment by tagomoris** [comment]
Tests of this change will fail because ArrayEventStream does NOT dup records correctly (and it's not tested in old test code).
Merging #1062 will solve this problem.

**Comment by repeatedly** [comment]
After merged this and https://github.com/fluent/fluentd/pull/1066, we should release v0.14.1 soon.
Currently, many users can't test fluentd v0.14 and new users from `gem install fluentd`may have a bad first impression.
This is not good for fluentd itself and community.

File: lib/fluent/plugin/out_roundrobin.rb [hunk 1]
```diff
@@ -14,68 +14,42 @@
...
     def initialize
       super
-
-      @outputs = []
       @weights = []
     end
 
-    attr_reader :outputs, :weights
+    attr_reader :weights
     attr_accessor :rand_seed
```
Comment (repeatedly) [inline lib/fluent/plugin/out_roundrobin.rb]: Can we remove this accessor?

Comment (tagomoris) [inline lib/fluent/plugin/out_roundrobin.rb]: Maybe yes. I have no idea about it.

**Comment by repeatedly** [comment]
LGTM with removing unnecessary `rand_seed`.

**Comment by tagomoris** [comment]
I pushed a comment to remove `rand_seed`.

**Comment by tagomoris** [comment]
Rebased on #1063 changes due to fixes on `event.rb`.

**Comment by tagomoris** [comment]
I've rebased on current master HEAD. I'll merge this change after CI green.

**Comment by tagomoris** [comment]
Merged!

**Pull request merged**